src/sentry/static/sentry/app/actionCreators/integrations.jsx

```diff
@@ -49,3 +50,100 @@ export function addIntegrationToProject(orgId, projectId, integration) {
     }
   );
 }
+
+/**
+ * Delete a respository
+ *
+ * @param {Object} client ApiClient
+ * @param {String} orgId Organization Slug
+ * @param {Number} repositoryId Repository ID
+ */
+export function deleteRepository(client, orgId, repositoryId) {
+  addLoadingMessage();
+  let promise = client.requestPromise(`/organizations/${orgId}/repos/${repositoryId}/`, {
+    method: 'DELETE',
+  });
+  promise.then(
+    () => clearIndicators(),
+    () => addErrorMessage(t('Unable to delete repository.'))
```

Comment:
Note that `addErrorMessage` will clear by default. Or did you want to explicitly clear for some other reason?

Reply:
I wanted to clear in the success case, so the 'loading' message doesn't stick around forever.